rfc-149-switch-to-per-page-asset-loading.md

```diff
@@ -0,0 +1,208 @@
+---
+Author: Ian James
+Date: 2 August 2022
+Deadline for decision: 16 August 2022
+---
+
+# Serve component CSS and JavaScript as individual files
+
+## Summary
+
+Component stylesheets and JavaScript should only be included for components present on a page on GOV.UK. This will lead to a reduction in page size, improve the ability of the browser to cache assets - which in turn improves performance for user journeys and return visits - and help reduce our reliance on manual asset auditing.
+
+The eleven applications that render GOV.UK use the [GOV.UK Publishing Components][govuk_publishing_components] gem. This gem is the single source of truth for shared components used across GOV.UK.
+
+Users that visit more than one page are likely to see pages rendered by different applications. Applications have unique but overlapping combinations of component use - this means that a user could download the same code in each application's concatenated stylesheets and JavaScript files when visiting pages rendered by different applications, despite the components being the same.
+
+Because each application concatenates and appends a fingerprint string to an asset's filename, even a small change to a component will cause the filename to change - meaning that return visitors will no longer have the advantage of caching.
+
+Currently asset auditing is a manual process that needs to be run locally. This means it's easy to miss when a component is no longer used, which leads to larger than needed stylesheets and JavaScript files being served to users.
```

Comment:
So the summary would just be the following and nothing else?

> Component stylesheets and JavaScript should only be included for components present on a page on GOV.UK. This will lead to a reduction in page size, improve the ability of the browser to cache assets - which in turn improves performance for user journeys and return visits - and help reduce our reliance on manual asset auditing.

Reply:
Done in https://github.com/alphagov/govuk-rfcs/pull/152/commits/b97a09a5957c7c70f46ca66e8eee4dc318a1c059.